netket/optimizer/sr.py

```diff
@@ -121,18 +155,25 @@ def __init__(
         self.qgt_kwargs = kwargs
         self.diag_shift = diag_shift
         self.diag_scale = diag_scale
+
+        check_conflicting_args_in_partial(
+            qgt,
+            ["diag_shift", "diag_scale"],
+            "The QGT arguments {} will be overwritten by the ones specified by SR (including defaults).",
```

Comment:
Let's make this warning a bit more explicit and actionable. 

```suggestion
            "Constructing the SR object with `SR(qgt= MyQGTType({}))` can lead to unexpected results and has been deprecated, "
            "because the keyword arguments specified in the QGTType are overwritten by those specified by the SR class and its defaults.\n\n"
            "To fix this, construct SR as  `SR(qgt=MyQGTType, {})` .\n\n"
            "In the future, this warning will become an error."
    ```